### PR TITLE
Feature/filter function

### PR DESCRIPTION
**Pull request by Jessica-Windoffer**

### CODE REVIEW

File: pages/activityList.js [hunk 1]
```diff
@@ -35,26 +36,127 @@ const StyledLi = styled.li`
   padding: 20px 0 20px 0;
 `;
 
+const StyledSection = styled.section`
+  display: flex;
+  flex-direction: column;
+  gap: 0.6rem;
+`;
```
Comment (benediktbrenk) [inline pages/activityList.js:43]: To have a better overview here use a seperate file for styled components.

File: pages/activityList.js [hunk 2]
```diff
@@ -35,26 +36,127 @@ const StyledLi = styled.li`
...
+    return activities.filter(
+      (activity) =>
+        activity.category.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        activity.title.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        activity.area.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        activity.country.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+  };
+
+  const filteredActivities = getFilteredActivities(searchTerm, activities);
```
Comment (benediktbrenk) [inline pages/activityList.js:100]: :)

File: pages/activityList.js [hunk 3]
```diff
@@ -35,26 +36,127 @@ const StyledLi = styled.li`
...
+            placeholder="Type to filter activities"
+            required
+          />
+        </WrapperSearchBar>
+      </StyledSection>
+      {filteredActivities && filteredActivities.length > 0 ? (
+        <StyledUl>
+          {filteredActivities.map((activity) => (
+            <>
+              <StyledLinkDetailsPage href={`/${activity.id}`}>
```
Comment (benediktbrenk) [inline pages/activityList.js:123]: For valid HTML <li> tags always have to be the direct child of an <ul>/<ol>/<menu> tag.
At the moment your list looks like this:

<ul>
  <a>
     <li></li>
  </a>
</ul>
But it has to be like this:

<ul>
  <li>
     <a></a>
  </li>
</ul>

Comment (benediktbrenk) [inline pages/activityList.js:123]: For valid HTML <li> tags always have to be the direct child of an <ul>/<ol>/<menu> tag.

**Review by benediktbrenk — CHANGES_REQUESTED** [review]
Good job :) Please see my comments in the code concerning the styled components and the ul/li parentage.

File: pages/activityList.js (hunk 1 above)
Comment (Jessica-Windoffer) [reply pages/activityList.js:43]: We think that is a really good suggestion and we definitely want to re-structure our styles and logic better in the future by outsourcing the styled components 😊 But we would like to do this as a separate User Story as we want to do this outsourcing in one step for every page/component 👍

File: pages/activityList.js [hunk 4]
```diff
@@ -35,26 +36,127 @@ const StyledLi = styled.li`
...
+  text-align: center;
+  margin: 0 0 25px 0;
+`;
+
 export default function ActivityList({ activities }) {
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleOnChange = (event) => {
+    setSearchTerm(event.target.value);
+  };
```
[the remark this answers is not in the document]
Comment (Jessica-Windoffer) [reply pages/activityList.js:84]: Done ✅

File: pages/activityList.js (hunk 3 above)
Comment (Jessica-Windoffer) [reply pages/activityList.js]: Changed the code to hopefully meet you commented improvement 😊

**Pull request merged**